### ISSUE TITLE
Expose `Constraint` in `cvxpy`

### ISSUE THREAD

**Issue by giuliobeseghi** [cvxpy/cvxpy#1881]
`Constraint` is not exposed to the public API. To import it, we have to do

```python
from cvxpy.constraints.constraint import Constraint
```

This is not the same of `Variable`, `Expression`, `Problem`, `Parameter`, where we can just do

```python
from cvxpy import Variable, Expression, Problem, Parameter
```

I wonder why that's the case? I use `Constraint` a lot, especially as a type hint when defining constraints. I'm wondering if it can be added to the `__init__` of `cvxpy`.

**Comment by giuliobeseghi** [comment]
similar issue for `Objective`, which I would for example use in a generic optimization class:

```python
class OptimizationModel:
    @abstractmethod
    def create_objective_function(...) -> Objective:
        """To be subclassed in specific optimization models. Returns Minimize/Maximize instances"""
```

**Comment by rileyjmurray** [comment]
You've touched on an important point. We need to do a better job of specifying what is and is-not in our public API. In our most recent developer call we decided that we'll use the global cvxpy namespace (things accessible from ``cvxpy.*``) for functions in the public API, as you suggest. However, there are functions in that namespace that we probably *dont* want in the public API. So I'd be hesitant to add things before we audit what's already in there. @cvxpy/maintainers what do you think?

**Comment by rileyjmurray** [comment]
This is resolved in CVXPY 1.3.
